icefoganalytics/travel-authorization
# Add CloudWatch Logging To App

Relates to:

- https://github.com/icefoganalytics/travel-authorization/issues/192
- https://github.com/icefoganalytics/wrap/commit/87267f5e90c03329d42c3ccf1e9208b9a7f61794

# Context

Currently the app logs are hard to get to, and this makes debugging production issues very difficult.
The solution is to log to a location where we can access the logs; in a nice to read format.

# Implementation

Use `morgan` for request logging and `winston` as the actual logger.
Add `RELEASE_TAG` and `ARG GIT_COMMIT_HASH` to back-end build pipeline.
Move all routers out of app.ts, and into "routes/index" file, so that app.ts is less cluttered.
I'm _not_ currently logging SQL statements in `db-client.ts` and `db-client-legacy.ts`.
Add `dev branch-from <github-issue-url>` dev helper (from ELCC -> IDP projects)

# Screenshots

New app logging
![image](https://github.com/icefoganalytics/travel-authorization/assets/23045206/f42cd2e6-ea5c-4ad5-b8e0-ff7c287fa5bd)

# Testing Instructions

1. Run the test suite via `dev test` (or `dev test_api`)
2. Boot the app via `dev up`
3. Log in to the app at http://localhost:8080
4. Log at the container logs and see that the logs are nicer.
